src/components/QuestionPlay/index.tsx

```diff
@@ -1,135 +1,156 @@
+  selectedOptionId: number | null | string;
   selectedQuestion: {
     questionId?: string;
     text: string;
     options: IOption[];
+    points: number;
+    questionNum?: number;
   };
-  isWithoutOptions: boolean;
   isAttempted: boolean;
 }
 
 export default function QuestionPlay({
   submitResponse,
-  onClose,
-  isPreview = false,
-  isQuestionSaved = true,
   pauseTimer,
   selectedOptionId,
   selectedQuestion,
-  isWithoutOptions,
   isAttempted,
 }: Props) {
-  const [selectedChoice, setSelectedChoice] = useState('');
-  const [isAnswerRevealed, setIsAnswerRevealed] = useState(isPreview);
-  const { options, questionId, text } = selectedQuestion;
+  const [selectedChoice, setSelectedChoice] = useState(selectedOptionId);
+  const [isAnswerRevealed, setIsAnswerRevealed] = useState(false);
+  const { options, text, points } = selectedQuestion;
+  const getQuestionTextStyles = (theme, isCorrect = false) => ({
+    backgroundColor: isCorrect ? theme.colors.green[2] : '#AFD0D4',
+    borderRadius: theme.radius.xs,
+  });
+  const { handleSubmit } = useForm();
 
   useEffect(() => {
-    if (isAttempted && !isPreview) {
+    if (isAttempted) {
       // reveal answer when question is attempted due to timeout
       setIsAnswerRevealed(true);
     }
-  }, [isAttempted, isPreview]);
+  }, [isAttempted]);
 
   useEffect(() => {
-    setIsAnswerRevealed(isPreview);
-  }, [questionId, isPreview]);
+    setSelectedChoice(selectedOptionId);
+  }, [selectedOptionId]);
 
-  function handleSubmit(ev: React.FormEvent) {
-    ev.preventDefault();
+  function submitQuestionResponse() {
+    document.getElementById('btnSubmitResponse')?.click();
+  }
+
+  function handleSelectedChoice(choice) {
+    setSelectedChoice(choice);
 
-    // submit response or save question on clicking submit or save button
-    submitResponse(selectedChoice);
+    setTimeout(() => {
+      submitQuestionResponse();
+    }, 100);
   }
 
   return (
-    <form className={styles.container} onSubmit={handleSubmit}>
-      <Markdown>{text}</Markdown>
-      <Divider />
-      <div className="flex flexWrap">
-        {isWithoutOptions
-          ? // Show correct answer for question without options when answer is revealed
-            isAnswerRevealed && (
-              <div className={styles.correctAns}>
-                <div className={styles.heading}>Correct answer</div>
+    <Card shadow="sm" p="lg" radius="md" my="sm" withBorder className="secondaryCard">
+      <form onSubmit={handleSubmit(() => submitResponse(selectedChoice === '' ? null : selectedChoice))}>
+        <Group>
+          <Title mr="xl" order={4}>
+            Question {selectedQuestion.questionNum}
+          </Title>
+          <Text weight="bold" component="span" size="sm">
+            Points: {points}
+          </Text>
+        </Group>
+        <Box my="xs" sx={getQuestionTextStyles}>
+          <Markdown>{text}</Markdown>
+        </Box>
+        {options.length === 1 ? (
```

Comment:
Can we store `const isWithoutOptions  = options.length === 1` so that the code is more readable?